lib/turn.rb

```diff
@@ -99,4 +100,30 @@ def check_vertical_win
     column_as_array = board_as_states.map { |row| row[@columns.index(@move)] }
     @player.winner = true if column_as_array.join.include?(winning_string)
   end
+
+  #range is acting as way to create index positions to check index positions on our board
+  def diag_offset_columns 
+    board = get_board_as_states
+    (1..3).map do |outer_int|
+      (outer_int..(board[0].size - 1)).collect {|inner_int| board[inner_int - outer_int][inner_int]}
+    end
+  end
+
+  def diag_offset_rows
+    board = get_board_as_states
+    (0..2).map do |outer_int|
+      (outer_int..(board.size - 1)).collect {|inner_int| board[inner_int][inner_int - outer_int]}
+    end
+  end
+
+  def check_diagonal_win
```

Comment:
When we rotate the board, we're going to need to be careful with how we do the ranges... the board is a 7x6, so when we rotate it, it's gonna be 6x7 instead. Just food for thought on the ranges.